modules/vaos/app/services/eps/appointment_service.rb

```diff
@@ -69,6 +76,27 @@ def submit_appointment(appointment_id, params = {})
 
     private
 
+    # Validates that there is no existing active appointment for the given referral.
+    # An appointment is considered active unless its status is either 'cancelled' or 'no-show'.
+    #
+    # @param referral_id [String] the ID of the referral to check for an existing appointment.
+    # @raise [Common::Exceptions::ValidationErrors] if an active appointment already exists.
+    def validate_no_existing_appointment!(referral_id)
+      existing_appointments = get_appointments
+      appointments = existing_appointments&.appointments || []
+
+      existing_appointment = appointments.find do |appt|
+        appt.referral_id == referral_id &&
```

Comment:
Note: FYI - EPS returns a referral_number which is then assigned to referral_id in its model.